generators/server/index.js

```diff
@@ -342,6 +342,130 @@ module.exports = class JHipsterServerGenerator extends BaseBlueprintGenerator {
         return this._writing();
     }
 
+    _postWriting() {
+        return {
+            packageJsonScripts() {
+                const packageJsonStorage = this.createStorage('package.json');
+                const packageJsonConfigStorage = packageJsonStorage.createStorage('config').createProxy();
+                packageJsonConfigStorage.backend_port = this.serverPort;
+                packageJsonConfigStorage.packaging = process.env.JHI_WAR === '1' ? 'war' : 'jar';
+                if (process.env.JHI_PROFILE) {
+                    packageJsonConfigStorage.default_environment = process.env.JHI_PROFILE.includes('dev') ? 'dev' : 'prod';
+                }
+            },
+            packageJsonDockerScripts() {
+                const packageJsonStorage = this.createStorage('package.json');
+                const scriptsStorage = packageJsonStorage.createStorage('scripts');
+                const databaseType = this.jhipsterConfig.databaseType;
+                if (databaseType === 'sql') {
+                    const prodDatabaseType = this.jhipsterConfig.prodDatabaseType;
+                    if (prodDatabaseType === 'no' || prodDatabaseType === 'oracle') {
+                        scriptsStorage.set(
+                            'docker:db',
+                            `echo "Docker for db ${prodDatabaseType} not configured for application ${this.baseName}"`
+                        );
+                    } else {
+                        scriptsStorage.set('docker:db', `docker-compose -f src/main/docker/${prodDatabaseType}.yml up -d`);
+                    }
+                } else {
+                    const dockerFile = `src/main/docker/${databaseType}.yml`;
+                    if (databaseType === 'cassandra') {
+                        scriptsStorage.set({
+                            'docker:db:await': 'wait-on tcp:9042 && sleep 20',
```

Comment:
that's the reason why I launch docker compose before tests

Reply:
@pascalgrimaud yes, it's something like this is written in the script file.
Old scripts:
- start every docker, except cassandra and couchbase, they are build only
- run frontend and backend tests.
- start cassandra and couchbase docker.
- run e2e.

New scripts:
- start every docker if not testcontainer (for e2e and backend).
- run frontend and backend test.
- start every docker if testcontainer (for e2e).
- run e2e.

Problems:
1 - testcontainer seems to be always enabled for cassandra.
Workaround: Enable testcontainer for every Cassandra build
2 - testcontainer is conflicting with docker. Probably due to port randomization.
Workaround: start docker after backend test.
3 - make sure backend isn't using docker instead of testcontainer.
Solution: start docker after backend test.
4 - there is no easy way to know if cassandra is ready.
Workaround: sleep.

Do you have any suggestion?
Fixing the problems above it's out of the scope of this PR.

Reply:
I'm fine with your change, it was just a comment, for your information, nothing more.

About sleep, Couchbase is impacted too: https://github.com/jhipster/generator-jhipster/blob/master/test-integration/scripts/24-tests-e2e.sh#L11
and should work as similar as Cassandra

Reply:
It’s done then. 😄